### PR TITLE
[@astrojs/rss] Quality-of-Life Improvement to `items` property-related error

### PR DESCRIPTION
**Pull request by cdvillard**
## Changes
This commit addresses a quality-of-life concern when setting up a RSS feed using collections. Specifically:

- it provides more context to the error message thrown when the object passed to the `items` property is missing any of the three required keys or if one of those keys is mistyped.
- it adds a link to provide additional context and instruction

## Testing

I tested this change similarly to how I encountered #9171. Using the `examples/blog` project, I simply messed around with the properties of `items` to ensure an error would render, and that my changes would build and reflect appropriately.

Before:
Screenshot of items error before my changes

After:
Screenshot of items error after my changes

## Docs

This shouldn't necessitate any changes to the docs as this way entirely done to help improve the error solely. Usage of the package remains unchanged.

### CODE REVIEW

**Comment by cdvillard** [comment]
Just to put it out there, I'm not married to the copy language used, nor the method. I was a little confused by the use of `errorMap` on line 105. I tried to define a new conditional there to catch `invalid_union` types, but despite building fine, it never seemed to catch it. I have no deep experience with Zod, so I'm sure it could be something specific to the `rssOptionsValidator` ZodObject alone.

File: .changeset/thirty-hairs-pump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/rss': patch
+---
+
+This commit addresses a quality-of-life concern when setting up a RSS feed using collections. Specifically, it provides more context to the error message thrown when the object passed to the `items` property is missing any of the three required keys or if one of those keys is mistyped.
```
Comment (sarah11918) [inline .changeset/thirty-hairs-pump.md:5]:
```suggestion
Improves the `@astrojs/rss` error message thrown when the object passed to the `items` property is missing any of the three required keys or if one of those keys is mistyped.
```
Appreciate the precision of the original statement! Just formatted according to our normal structure (starting with the verb) so the reader will hit the actual PR contributions right away.

**Review by sarah11918 — COMMENTED** [review]
A docs review only: the proposed error message is fantastic. Thank you for the care and attention to detail @cdvillard! Massaged the changeset a little, so see what you think about that.

File: .changeset/thirty-hairs-pump.md (hunk 1 above)
Comment (cdvillard) [reply .changeset/thirty-hairs-pump.md]: @sarah11918 Happy to take the suggestion! Simplifies it to exactly what's necessary.

**Review by ematipico — APPROVED** [review]

**Pull request merged**